### DISCUSSION TITLE
Task timeouts are not working as expected

### DISCUSSION THREAD

**Discussion by saikotcharlakota** [apache/airflow#35967] [Q&A] [answered]
I have a `PythonOperator` with task timeout set to 3hrs. 
However, the task kept running for over 11 hours. I am not sure if this is a bug or an issue with our configuration. Any help is much appreciated.

**Environment:**
Airflow v2.5.1
Python 3.10

![Screenshot 2023-11-29 at 11 55 28 AM](https://github.com/apache/airflow/assets/96261216/3b166a37-92cb-420b-b5e7-1f57aa81025e)
![Screenshot 2023-11-29 at 11 58 25 AM](https://github.com/apache/airflow/assets/96261216/609bd636-b16a-43e3-bf69-e50a7df19073)

**Task Config:**
```
send_email_with_processed_data = PythonOperator(
    task_id="send_email_with_processed_data",
    python_callable=process_data,
    dag=dag,
    provide_context=True,
    execution_timeout=timedelta(hours=3),
)

**Comment by potiuk** [comment, marked as answer]
There are cases where timeout will not work - mostly depending on some low-level code running in C for a long time, or when your system resources are used up or when your code catches and ignores all exceptions.

What is your case - hard to say - but you should read  https://github.com/apache/airflow/issues/35474 and https://github.com/apache/airflow/issues/35644 for details and some possible remedies that you can apply. other than that - we have an idea what we could improve to make it more robuts (see in the issues) but those ideas are not implemented yet.

**Reply by saikotcharlakota** [reply]
Thanks for shedding light here.
